examples/05_vol_geom_wasm_threejs/index.html

```diff
@@ -0,0 +1,330 @@
+			var normals_copied = Module.ccall('frame_normals_copied', 'number');
+			var normals_sz = Module.ccall('frame_normals_sz', 'number');
+			var vn_f32 = new Float32Array(Module.HEAP8.buffer, normals_copied, normals_sz / 4);
+			vologram.geometry.setAttribute( 'normal', new THREE.Float32BufferAttribute(vn_f32, 3 ));
+		}
+
+		// Key-Frames also contain texture coordinate and index data.
+		if (is_key) {
+			vologram.last_keyframe_loaded = frame_idx;
+
+			// Texture Coordinates - fetch and upload.
+			var uvs_copied = Module.ccall('frame_uvs_copied', 'number');
+			var uvs_sz = Module.ccall('frame_uvs_sz', 'number');
+			var uvs_f32 = new Float32Array(Module.HEAP8.buffer, uvs_copied, uvs_sz / 4);
+			vologram.geometry.setAttribute( 'uv', new THREE.Float32BufferAttribute( uvs_f32, 2 ));
+
+			// Indices - fetch and upload.
+			var indices_copied = Module.ccall('frame_indices_copied', 'number');
+			var indices_sz = Module.ccall('frame_i_sz', 'number');
+			vologram.n_indices = indices_sz / 2; // ushort
+			var indices_u16 = new Uint16Array(Module.HEAP8.buffer, indices_copied, vologram.n_indices);
+			vologram.geometry.setIndex( new THREE.Uint16BufferAttribute( indices_u16, 1 ))
+		}
+
+		// It seems that calculating bounding sphere does not work and always returns NaN value for the radius
+		if (vologram.geometry.boundingSphere === null)
+			vologram.geometry.boundingSphere = new THREE.Sphere();
+		if (
+			vologram.geometry.boundingSphere.radius === 0 ||
+			isNaN(vologram.geometry.boundingSphere.radius)
+		) {
+			vologram.geometry.computeBoundingBox();
+			vologram.geometry.boundingBox.getBoundingSphere(vologram.geometry.boundingSphere);
+		}
+
+		vologram.last_frame_loaded = frame_idx;
+		return true;
+	}
+
+	function start_video() {
+		var texture_video_el = document.getElementById("texture_video_el");
+		texture_video_el.src = vologram.video_url;
+		texture_video_el.requestVideoFrameCallback(doSomethingWithTheFrame);
+		texture_video_el.play();
+		console.log("video started");
+	}
+
+	function init_vologram() {
+		if (!vologram.header_ready || !vologram.sequence_ready || vologram.ready) { return; } // Wait until both files are downloaded.
+		var ret = Module.ccall('create_file_info', 'boolean', ['string', 'string'], ['header.vols', 'sequence_0.vols']);
+		console.log('create_file_info = ' + ret);
+		if (!ret) {
+			console.error("failed to load vologram");
+			return;
+		}
+		vologram.has_normals = Module.ccall('has_normals', 'boolean');
+		vologram.max_blob_size = Module.ccall('max_blob_sz', 'number')
+		console.log( "max blob size =", vologram.max_blob_size );
+		create_vologram();
+		console.log("vologram.has_normals = " + vologram.has_normals);
+		mesh_from_frame(0); // Load in frame 0 so something displays even if we don't play yet.
+		console.log("mesh initialised");
+		start_video();
+		vologram.ready = true;
+	}
+
+	// This can be called to add an artificial slow-down in the code. Parameter value is in micro-seconds.
```

Comment:
if we call this function in different places when testing we can test the robustness vs desync.

Reply:
I left it in for that. Could be handy in the future for unit/fuzz testing